packages/@aws-cdk/aws-codecommit/test/codecommit.test.ts

```diff
@@ -66,6 +70,91 @@ describe('codecommit', () => {
 
     });
 
+    test('can be setup with file deployment', () => {
+      // GIVEN
+      const app = new App({
+        context: {
+          [cxapi.DISABLE_ASSET_STAGING_CONTEXT]: 'true',
+        },
+      });
+      const stack = new Stack(app, 'MyStack');
+
+      // WHEN
+
+      new Repository(stack, 'Repository', {
+        repositoryName: 'MyRepositoryName',
+        code: Code.fromZipFile(join(__dirname, 'asset-test.zip')),
+      });
+
+      // THEN
+      const entry = stack.node.metadataEntry.find(m => m.type === 'aws:cdk:asset');
+      expect(entry).toBeTruthy();
+
+      // verify that now the template contains parameters for this asset
+      const assembly = app.synth();
+      const template = JSON.parse(readFileSync(join(assembly.directory, 'MyStack.template.json'), { encoding: 'utf-8' }));
+
+      expect(template.Resources.Repository22E53BBD.Properties.Code.S3.Bucket).toBeDefined();
+    });
+
+    test('can be setup with directory deployment', () => {
+      // GIVEN
+      const app = new App({
+        context: {
+          [cxapi.DISABLE_ASSET_STAGING_CONTEXT]: 'true',
+        },
+      });
+      const stack = new Stack(app, 'MyStack');
+
+      // WHEN
+
+      new Repository(stack, 'Repository', {
+        repositoryName: 'MyRepositoryName',
+        code: Code.fromDirectory(join(__dirname, 'asset-test')),
+      });
+
+      // THEN
+      const entry = stack.node.metadataEntry.find(m => m.type === 'aws:cdk:asset');
+      expect(entry).toBeTruthy();
+
+      // verify that now the template contains parameters for this asset
+      const assembly = app.synth();
+      const template = JSON.parse(readFileSync(join(assembly.directory, 'MyStack.template.json'), { encoding: 'utf-8' }));
+
+      expect(template.Resources.Repository22E53BBD.Properties.Code.S3.Bucket).toBeDefined();
+    });
+
+    test('can be setup with asset deployment', () => {
+      // GIVEN
+      const app = new App({
+        context: {
+          [cxapi.DISABLE_ASSET_STAGING_CONTEXT]: 'true',
+        },
+      });
+      const stack = new Stack(app, 'MyStack');
+
+      const readmeAsset = new Asset(stack, 'ReadmeAsset', {
+        path: join(__dirname, 'asset-test'),
+      });
+
+      // WHEN
+
+      new Repository(stack, 'Repository', {
+        repositoryName: 'MyRepositoryName',
+        code: Code.fromAsset(readmeAsset),
+      });
+
+      // THEN
+      const entry = stack.node.metadataEntry.find(m => m.type === 'aws:cdk:asset');
+      expect(entry).toBeTruthy();
+
+      // verify that now the template contains parameters for this asset
+      const assembly = app.synth();
+      const template = JSON.parse(readFileSync(join(assembly.directory, 'MyStack.template.json'), { encoding: 'utf-8' }));
+
+      expect(template.Resources.Repository22E53BBD.Properties.Code.S3.Bucket).toBeDefined();
+    });
+
```

Comment:
Missing negative test for using a non-directory in `Code.fromDirectory()`, and a directory in `Code.fromZipFile()`.